src/aleph/web/controllers/storage.py

```diff
@@ -56,20 +92,133 @@ async def add_storage_json_controller(request: web.Request):
     return web.json_response(output)
 
 
+async def _verify_message_signature(
+    pending_message: BasePendingMessage, chain_service: ChainService
+) -> None:
+    try:
+        await chain_service.verify_signature(pending_message)
+    except InvalidSignature:
+        raise web.HTTPForbidden()
+
+
+async def _verify_user_balance(
+    pending_message_db: PendingMessageDb, session: DbSession, size: int
+) -> None:
+    current_balance = get_total_balance(
+        session=session, address=pending_message_db.sender
+    ) or Decimal(0)
+    required_balance = (size / MiB) / 3
+    current_cost_for_user = get_total_cost_for_address(
+        session=session, address=pending_message_db.sender
+    )
+    if current_balance < (Decimal(required_balance) + current_cost_for_user):
+        raise web.HTTPPaymentRequired
```

Comment:
Pretty sure you need to build the object.

```suggestion
        raise web.HTTPPaymentRequired()
```